card/mod/core/set/all/fetch_helper.rb

```diff
@@ -46,21 +53,16 @@ def skip_type_lookup? opts
     opts[:skip_virtual] || opts[:new].present? || opts[:skip_type_lookup]
   end
 
+  # @return [{Card}, {True/False}] Card object and "needs_caching" ruling
   def retrieve_existing mark, opts
     return [nil, false] unless mark.present?
     mark_type, mark_key = retrievable_mark_type_and_value mark
-    needs_caching = false # until proven true :)
-
-    # look in cache
     card = send "retrieve_from_cache_by_#{mark_type}", mark_key, opts[:local_only]
-
-    if retrieve_from_db? card, mark_type, mark_key, opts[:look_in_trash]
-      # look in db if needed
-      card = retrieve_from_db mark_type, mark_key, opts[:look_in_trash]
-      needs_caching = !card.nil? && !card.trash
+    if card&.real?
+      [card, false]
+    else
+      retrieve_from_db card, mark_type, mark_key, opts
```

Comment:
:alarm: :alarm: :alarm: I refactored fetch again :(

I tried very hard not to change the logic.  There's not a great reason for this, except I suppose the long term code climate goals and general readability needs.

Reply:
tests for retrieve_existing?  fair enough.

Reply:
can I make a separate chore story for that?